src/UserInterface/CarMap.py

```diff
@@ -111,15 +151,20 @@ def check_vehicle_proximity(self, vehicle_id: str, position: dict,) -> None:
         position : dict
             Dictionary containing the 'x' and 'y' coordinates of the vehicle's position in the simulation.
         """
-        for v in self._environment_manager._active_virtual_cars:
-            if v == vehicle_id:
-                self.check_virtual_vehicle_proximity(vehicle_id, position)
+        if self.__can_virtual_cars_hack is True:
+            for v in self._environment_manager._active_virtual_cars:
+                if v == vehicle_id:
+                    self.check_hack_vehicle_proximity(vehicle_id, position)
+        if self.__can_physical_cars_hack is True:
+            for v in self._environment_manager._active_physical_cars:
+                if v == vehicle_id:
+                    self.check_hack_vehicle_proximity(vehicle_id, position)
```

Comment:
No need to write "is True". e.g. l.154 could just be "if self.__can_virtual_cars_hack:"

Reply:
self.__can_..._cars_be_hacked should be checked here, shouldn't it?